content/articles/introduction-to-evolutionary-algorithms/index.md

```diff
@@ -0,0 +1,431 @@
+
+![Phases-of-ea](/engineering-education/introduction-to-evolutionary-algorithms/Phases.png)
+
+To delve deeper into the aspects of EA, let us consider a problem with monkeys, our evolutionary ancestors.
+
+### The Infinite Monkey Theorem
+>The infinite monkey theorem states that a monkey hitting keys at random on a typewriter keyboard for an infinite amount of time will almost surely type any given text, such as the complete works of William Shakespeare.
+
+The actual chance that a monkey types the entire Julius Caesar is very slim. It is so minuscule that if it started at Big Bang, it would still be typing. This is going to set the premise for the problem we are going to look at.
+
+Let us consider a monkey aiming to type a short phrase, say "Hello".
+The probability that the monkey gets one letter right, say "H", would be 1/54 as there are 27 possible lowercase and 27 possible uppercase characters to choose from. So, the probability that the monkey gets the word 'Hello' right would be:
+> <sup>1</sup>/<sub>54<sup>5</sup></sub>
+
+= 1 in 459165024
+
+If our monkey is a powerful simulation that can type a million characters a second, this is going to take about 7.6 minutes.
+7.6 minutes is not a great deal of time. If that's the case, why do we still need EA?
+
+### Why do we need Evolutionary Algorithms?
+We considered the word "Hello", now let us look at the phrase "HelloWorld"
+The probability now becomes 1 in 54<sup>10</sup> which is:
+> 1 in 2,10,83,25,19,26,49,20,576
+
+With a simulation that types a million words in a second, this is going to take:
+> 6685.455 years.
+
+Can you still say that it doesn't seem like a big enough time? When a mere increase of 5 more letters takes more than 100 lifetimes, then what about full sentences?
+
+Let us see what happens when we plug the same target "HelloWorld" in an Evolutionary Algorithm. It gives the result in approximately **1 second** and 165 iterations.
+
+![Output](/engineering-education/introduction-to-evolutionary-algorithms/HelloWorld.PNG)
+
+Let us see how an Evolutionary Algorithm manages to speed things up by such a massive amount.
+
+### Coding an Evolutionary Algorithm
+We are going to try to make our monkey type the word "Hello" to learn the finer workings of an EA.
+
+#### Step 1: Initialization
+Ideally, the initial set of solutions for our problem should have all the possible 5 letter words. But, this set is going to be of length 459165024. We do not need to keep every single value possible in our population. We will be creating new genetic material. Let us take our population size to be 50.
+
+So, let us create an array with 50 random strings of upper and lower case alphabets of length 5 [Equal to our target string]. We will be using the random.choices() method to get our random strings.
+
+````python3
+import string
+import random
+target = "HelloWorld"
+n = 50
+population = [''.join(random.choices(string.ascii_letters, k=len(target))) for _ in range(n)]
+````
```

Comment:
@Tvashta Shouldn't the target string here be "Hello" rather than "HelloWorld"?

Reply:
You are right. Sorry! Updated now